index-scheduler/src/lib.rs

```diff
@@ -1262,6 +1270,24 @@ impl IndexScheduler {
 
         wtxn.commit().map_err(Error::HeedTransaction)?;
 
+        // Once the tasks are commited, we should delete all the update files associated ASAP to avoid leaking files in case of a restart
+        tracing::debug!("Deleting the upadate files");
+
+        ids.into_par_iter().try_for_each(|id| -> Result<()> {
+            let rtxn = self.read_txn()?;
```

Comment:
it would be better to create the read txn once (using the appropriate feature). Opening a lot of transactions in parallel can be slower than using the same one on different threads. What do you think?

Reply:
How would I do that? By using the `sync-read-txn` heed feature?

Because yeah, I wasn’t happy about this solution, I thought of using [`for_each_init`](https://docs.rs/rayon/latest/rayon/iter/trait.ParallelIterator.html#method.for_each_init) with this code:
```rust
...
        ids.into_par_iter().try_for_each_init(
            || self.read_txn(),
            |rtxn, id| -> Result<()> {
                // let rtxn = self.read_txn()?;
                let rtxn = rtxn.as_ref().unwrap();
                let task = self
                    .get_task(rtxn, id)
...
```
But from what I understand, it pretty much ends up calling my closure for every rayon job (thus every iteration, right?).

Reply:
The other easy idea could be to split up the roaring bitmap into as many parts as we have threads and then let each thread open **one** `rtxn` and delete the batch of files on his side.
That’s probably better, but if one finishes early, the other will be waiting.

We could maybe fix that with an AtomicU64 + the `.select` method on the roaring bitmap as we did in Arroy. That seems a bit overkill, but tbh, it’s not that complex to write, and it works.

But let me know if you have anything else in mind

Reply:
I implemented the last solution